src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -476,10 +515,156 @@ static bool CellIsNull(ChunkCollection &collection, idx_t column, idx_t index) {
+                                 const idx_t order_end, ChunkCollection &boundary, const idx_t boundary_row) {
+	D_ASSERT(!CellIsNull(boundary, 0, boundary_row));
+	const auto val = GetCell<T>(boundary, 0, boundary_row);
+
+	OperationCompare<T, OP> comp;
+	ChunkCollectionIterator<T> begin(over, order_col, order_begin);
+	ChunkCollectionIterator<T> end(over, order_col, order_end);
+	if (FROM) {
+		return idx_t(std::lower_bound(begin, end, val, comp));
+	} else {
+		return idx_t(std::upper_bound(begin, end, val, comp));
+	}
+}
+
+template <typename OP, bool FROM>
+static idx_t FindRangeBound(ChunkCollection &over, const idx_t order_col, const idx_t order_begin,
+                            const idx_t order_end, ChunkCollection &boundary, const idx_t expr_idx) {
+	const auto &over_types = over.Types();
+	D_ASSERT(over_types.size() > order_col);
+	D_ASSERT(boundary.Types().size() == 1);
+	D_ASSERT(boundary.Types()[0] == over_types[order_col]);
+
+	switch (over_types[order_col].InternalType()) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		return FindTypedRangeBound<int8_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::INT16:
+		return FindTypedRangeBound<int16_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::INT32:
+		return FindTypedRangeBound<int32_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::INT64:
+		return FindTypedRangeBound<int64_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::UINT8:
+		return FindTypedRangeBound<uint8_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::UINT16:
+		return FindTypedRangeBound<uint16_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::UINT32:
+		return FindTypedRangeBound<uint32_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::UINT64:
+		return FindTypedRangeBound<uint64_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::INT128:
+		return FindTypedRangeBound<hugeint_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::FLOAT:
+		return FindTypedRangeBound<float, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::DOUBLE:
+		return FindTypedRangeBound<double, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	case PhysicalType::INTERVAL:
+		return FindTypedRangeBound<interval_t, OP, FROM>(over, order_col, order_begin, order_end, boundary, expr_idx);
+		break;
+	default:
+		throw InternalException("Unsupported column type for RANGE");
```

Comment:
Can you add a test that tries to perform RANGE on e.g. a VARCHAR or LIST column (and triggers an exception)?

Remember that an InternalException always results in a test failure, so this might need to be changed to a different exception if it is possible to trigger this.

Reply:
RANGE queries like that get caught earlier (during expression translation - see the VARCHAR test I added above). This really is an exception that should not happen.